fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/DynamicRegistriesImpl.java

```diff
@@ -71,8 +136,8 @@ public static <T> DynamicRegistries.Settings<T> register(RegistryKey<? extends R
 		}
 
 		var entry = new RegistryLoader.Entry<>(key, codec);
-		DYNAMIC_REGISTRIES.add(entry);
-		sorted = false;
+		// TODO: may not be thread-safe
```

Comment:
Does it need to be? I think registrations should only happen from mod initialisers which are single-threaded on Fabric Loader.

Reply:
Indeed, probably overkill.